src/quicktype-core/language/CPlusPlus.ts

```diff
@@ -479,9 +481,18 @@ export class CPlusPlusRenderer extends ConvenienceRenderer {
 
     protected emitEnum(e: EnumType, enumName: Name): void {
         const caseNames: Sourcelike[] = [];
-        this.forEachEnumCase(e, "none", name => {
+        const enumValues = enumCaseNames(e, this.targetLanguage.name );
+
+        this.forEachEnumCase(e, "none", (name, jsonName) => {
             if (caseNames.length > 0) caseNames.push(", ");
             caseNames.push(name);
+
+            if (enumValues !== undefined) {
+                const [ enumvalue, isFixed ] = getAccessorName(enumValues, jsonName);
+                if (enumvalue !== undefined && isFixed !== undefined) {
```

Comment:
`isFixed` is a boolean.  The comparison to `undefined` will always succeed.

The fixed thing only really applied to names - if a name is given for a specific target language then it's taken verbatim.